site/zenodo_rdm/subcommunities/request.py

```diff
@@ -113,8 +124,180 @@ class ZenodoSubCommunityRequest(SubCommunityRequest):
+            "All records will be subject to the EU Open Research Repository "
+            "<a href='https://zenodo.org/communities/eu/curation-policy'>curation policy</a>. "
+            "For instance, you can only deposit records in the community related to the EU-funded "
+            "project</li><li><b>Verified:</b> All EU project communities are marked with a "
+            "Verified community badge</li></ul></b></br>The EU Open Research Repository is gradually "
+            "being improved and by mid-2025 new submissions will automatically be checked "
+            "for compliance with the related open science requirements in the Horizon Europe "
+            "grant agreement. For more information see <a href='https://zenodo.org/communities/eu/pages/join'>"
+            "https://zenodo.org/communities/eu/pages/join</a>.</br></br><b>Which EU-funded projects "
+            "have already joined?</b></br>You can browse the "
+            "<a href='https://zenodo.org/communities/eu/browse/subcommunities'>EU-funded projects</a> "
+            "which have already already joined.</br></br><b>When should I decline the invitation "
+            "to join?</b></br>You should <b>decline</b> this invitation if your Zenodo community "
+            "is not related to the above mentioned EU-funded project, or if the community is used "
+            "for multiple purposes (e.g both an organisation and a project).</br></br><b>"
+            "Further questions?</b></br>Don't hesitate to get in <a href='https://zenodo.org/support'>"
+            "touch with us</a> if you have any questions.</br></br>The request will be automatically "
+            f"accepted on <b>{expires_at}</b> in case you do not accept or decline the request by then."
+            "</br></br>Your sincerely,</br>The Zenodo team"
+        )
+
+        super().execute(identity, uow)
+
+        uow.register(
+            NotificationOp(
+                notifications.SubComInvitationCreate.build(
+                    identity=identity, request=self.request
+                )
+            )
+        )
+
+
+class SubcommunityInvitationExpireAction(actions.ExpireAction):
+    """Expire action."""
+
+    def _get_community_records(self, community_id):
+        """Get the records of a community."""
+        return current_community_records_service.search(
+            system_identity, community_id=community_id
+        )
+
+    def execute(self, identity, uow):
+        """Execute expire action."""
+        child = self.request.receiver.resolve().id
+        parent = self.request.created_by.resolve().id
+
+        # Move records
+        records = self._get_community_records(child)
+        current_rdm_records.record_communities_service.bulk_add(
+            system_identity, parent, (x["id"] for x in records), uow=uow
+        )
+
+        current_communities.service.bulk_update_parent(
+            system_identity, [child], parent_id=parent, uow=uow
+        )
```

Comment:
minor: this can be DRYed out a bit, since it looks like there's some repetition (that can also be faulty, as we saw above).